src/Algebra/Linear/Properties/VectorSpace.agda

```diff
@@ -0,0 +1,191 @@
+vgen-cong {f₁} {f₂} []       f₁≗f₂ = Setoid.reflexive ≈ᴹ-setoid Eq.refl
+vgen-cong {f₁} {f₂} (x ∷ xs) f₁≗f₂ = begin⟨ ≈ᴹ-setoid ⟩
+  f₁ x *ₗ x +ᴹ vgen f₁ xs ≈⟨ +ᴹ-congʳ (*ₗ-congʳ (f₁≗f₂ x)) ⟩
+  f₂ x *ₗ x +ᴹ vgen f₁ xs ≈⟨ +ᴹ-congˡ (vgen-cong xs f₁≗f₂) ⟩
+  f₂ x *ₗ x +ᴹ vgen f₂ xs ∎
+
+------------------------------------------------------------------------
+-- Some consequences of `VectorSpace` inner product properties.
+v∙g[x]+y-cong₂ : {g : V → V} {v x w : V} {y z : S} →
+                 Congruent _≈ᴹ_ _≈ᴹ_ g → x ≈ᴹ w → y ≈ z →
+                 v ∙ g x + y ≈ v ∙ g w + z
+v∙g[x]+y-cong₂ {g} {v} {x} {w} {y} {z} g-cong x≈w y≈z = begin⟨ setoid ⟩
+  v ∙ g x + y ≈⟨ +-congʳ (∙-congˡ (g-cong x≈w)) ⟩
+  v ∙ g w + y ≈⟨ +-congˡ y≈z ⟩
+  v ∙ g w + z ∎
+
+foldr-cong : ∀ {f g : V → S → S} {d e : S} →
+             (∀ {y z} → y ≈ z → ∀ x → f x y ≈ g x z) → d ≈ e →
+             foldr f d ≗ foldr g e
+foldr-cong f≈g d≈e []       = d≈e
+foldr-cong f≈g d≈e (x ∷ xs) = f≈g (foldr-cong f≈g d≈e xs) x
+
+foldr-homo-∙ : {v x₀ : V} {g : V → V} → Congruent _≈ᴹ_ _≈ᴹ_ g →
+               (xs : List V) →
+               v ∙ foldr (_+ᴹ_ ∘ g) x₀ xs ≈
+                 foldr (_+_ ∘ (v ∙_) ∘ g) (v ∙ x₀) xs
+foldr-homo-∙ _ [] = ∙-congˡ (≈ᴹ-reflexive Eq.refl)
+foldr-homo-∙ {v} {x₀} {g} g-cong (x ∷ xs) = begin⟨ setoid ⟩
+  v ∙ (g x +ᴹ foldr (_+ᴹ_ ∘ g) x₀ xs)        ≈⟨ ∙-distrib-+ ⟩
+  v ∙ g x + v ∙ foldr (_+ᴹ_ ∘ g) x₀ xs       ≈⟨ +-congˡ (foldr-homo-∙ g-cong xs) ⟩
+  foldr (_+_ ∘ (v ∙_) ∘ g) (v ∙ x₀) (x ∷ xs) ∎
+
+u∙-homo : ∀ {u} → IsModuleHomomorphism (u ∙_)
+u∙-homo = record
+  { isBimoduleHomomorphism = record
+      { +ᴹ-isGroupHomomorphism = record
+          { isMonoidHomomorphism = record
+              { isMagmaHomomorphism = record
+                  { isRelHomomorphism = record
+                      { cong = ∙-congˡ
+                      }
+                  ; homo = λ x y → ∙-distrib-+
+                  }
+              ; ε-homo = ∙-idʳ
+              }
+          ; ⁻¹-homo = λ x → ∙-homo-⁻¹
+          }
+      ; *ₗ-homo = λ r x → ∙-comm-*ₗ
+      ; *ᵣ-homo = λ r x → ∙-comm-*ᵣ
+      }
+  }
+
+------------------------------------------------------------------------
+-- Properties of linear maps from vectors to their underlying scalars.
+module _ (lm : LinearMap mod ⟨module⟩) where
+
+  open LinearMap lm
+  open ModHomo   mod ⟨module⟩ (LinearMap.homo lm)
+
+  vred : (V → S) → List V → S
+  vred g = foldr (_+_ ∘ uncurry _*_ ∘ < g , f >) 0#
+
+  foldr-homo : (g : V → S) → (xs : List V) → f (vgen g xs) ≈ vred g xs
+  foldr-homo g []       = 0ᴹ-homo
+  foldr-homo g (x ∷ xs) = begin⟨ setoid ⟩
+    f (h x (foldr h 0ᴹ xs))
+      ≈⟨ +ᴹ-homo (g x *ₗ x) (foldr h 0ᴹ xs) ⟩
+    f (g x *ₗ x) + f (foldr h 0ᴹ xs)
+      ≈⟨ +-congʳ (*ₗ-homo (g x) x) ⟩
+    g x * f x + f (foldr h 0ᴹ xs)
+      ≈⟨ +-congˡ (foldr-homo g xs) ⟩
+    g x * f x + vred g xs
+      ∎
+    where
+    h = _+ᴹ_ ∘ vscale g
+
+  vSum : List V → V
+  vSum xs = foldr _+ᴹ_ 0ᴹ xs
+
+  fScale : V → V
```

Comment:
What does the `f` stand for in the names here? Again comments on these functions would be welcome!

Reply:
`f` is the linear function contained within the linear map.
I've added a comment to this effect.

I've also made a clean-up pass through this code:

- eliminating any unused code,
- narrowing the scope of helper functions, and
- adding comments about intent/function.

**==> All changes described above arriving with next push.**